### ISSUE TITLE
Strange preferences of Named Expression names

### ISSUE THREAD

**Issue by YuriKirilin** [handsontable/hyperformula#880]
### Description

Adding Named Expression by hfInstance.addNamedExpression raises a error when the name ends with a number and there are no digits in the middle.

### Steps to reproduce
E.g. adding named expression "Product1" raises the error, while "Line2Product1" does not. From an end-user point of view, these names are similar, and Excel supports both of them.

### Your environment
* HyperFormula version: 1.3.0

### Hint
See the strange regular expression in NamedExpressions.js line 243 (isNameValid function)

**Comment by MartinDawson** [comment]
The issue is here:

```js
  public isNameValid(expressionName: string): boolean {
    if (/^[A-Za-z]+[0-9]+$/.test(expressionName)) {
      return false
    }
    return /^[A-Za-z\u00C0-\u02AF_][A-Za-z0-9\u00C0-\u02AF._]*$/.test(expressionName)
  }
```

In this commit: e1d91aa74ccda49b94219c4d1617e3f4c35032f7

This regex is wrong: `/^[A-Za-z]+[0-9]+$/.test(expressionName)`

Not even sure why it's doing this small regex here before the longer one.

**Comment by YuriKirilin** [comment]
For me, the regex from the spec looks too inaccurate as well, and it may be amended in the next spec versions.

I guess, this is not the only point where Excel diverges from various specifications. And a HyperFormula user, I'd prefer to have an option specifying strictness/compatibility mode to be able to use my existing datasheets with as few modifications as possible.

**Comment by krzysztofspilka** [comment]
@YuriKirilin thanks for raising this issue. Correct me if I'm wrong, but it's already mentioned in the docs: https://handsontable.github.io/hyperformula/guide/named-ranges.html#naming-convention

R1C1 | Incorrect
-- | --

But sure, I guess we could add to the [know limitations](https://handsontable.github.io/hyperformula/guide/known-limitations.html#google-sheets-and-microsoft-excel) section although I'm not entirely sure it's the right place for outlining discrepancies between different spreadsheet software. I wouldn't call that a limitation; it's a way we decided this should be implemented in the first place.

**Comment by sequba** [comment]
This bug is closely related to #1058. Both issues will be addressed in pull request #1130

**Comment by sequba** [comment]
@YuriKirilin,
 
I'm more than happy to announce that we just released [HyperFormula 2.3.1](https://github.com/handsontable/hyperformula/releases/tag/2.3.1) where this issue is fixed. We are closing this issue as solved. If there is anything that won't work for you after updating, please leave a comment.
